KATA-681: Understanding section: moving doc from Google Docs to GitHub (#32307)
openshift/openshift-docs: modules/sandboxed-containers-common-terms.adoc

//Module included in the following assemblies:
//
// * sandboxed_containers/understanding_sandboxed_containers.adoc
[id="sandboxed-containers-common-terms_{context}"]
= {sandboxed-containers-first} common terms

The following terms are used throughout the documentation.

> Sandbox:: A sandbox is a controlled environment where programs can run. Within the controlled environment, you can run untested or untrusted programs without risking harm to the host machine or the operating system. A sandbox keeps your workloads isolated from the rest of the system.

Reviewer: ```suggestion
Sandbox:: A sandbox is an isolated environment where programs can run. In a sandbox, you can run untested or untrusted programs without risking harm to the host machine or the operating system.
```

+

> In the context of {sandboxed-containers-first}, sandboxing is achieved by running workloads in a different kernel using virtualization, providing enhanced control over the interactions between multiple workloads that run on the same host.

Reviewer: Different from what? 

Consider breaking into two sentences.

> Pod:: A pod is a Kubernetes and {product-title} construct that represents resources where containers can be deployed. Containers run inside of pods, and pods are used to specify resources that can be shared between multiple containers.

Reviewer: Could this definition compare and contrast pods (general) vs. pods (this document)? 

Also, do we need to call out that it's both k8s + OCP, or can k8s be dropped? Not sure, myself. :)

Reviewer 2: Since OCP is not introducing a `pod` resource on its own and is in fact a rebase of Kubernetes (using it underneath), we initially thought it's okay to use kuberetes. It might be easier to reference an existing definition in OpenShift documentation to prevent though contention to map to a project vs. a product.

Reviewer 3: I understand @zanetworker's rationale, but I think that it does not hurt to write "A pod is a Kubernetes, OpenShift and {product-title} construct...". It does not seem overly verbose.

Or alternatively, "A pod is a {product-title} construct inherited from Kubernetes and OpenShift that represent resources where containers can be deployed." This wording may help clarify the points made in the next sentence that we are explaining how they work in the case of sandboxed containers.

+
In the context of {sandboxed-containers-first}, a pod is implemented as a virtual machine. Several containers can run in the same pod on the same virtual machine.

> {sandboxed-containers-operator}:: An Operator is a software component that automates operations, which are actions that a human operator could do on the system.

Reviewer: Could this be a brief concept module instead of a term?

Author: @maxwelldb The Operator itself as a building blocks section. Do you think we should move this content there instead? 
@zanetworker thoughts?

Reviewer 2: @maxwelldb the idea of introducing terms here is to make the user quickly familiar with concepts necessary to understand the product flow by consuming content from the same place.

Reviewer 3: @maxwelldb I think the difference with the example you gave us is the much smaller amount of configuration and tuning we expect in the case of sandboxed containers. At the moment at least, I think if we created a concept module, it would be mostly empty.

@zanetworker Do you think this could change in the future?

Reviewer 4: I haven't seen this succinct of a definition of our term `Operator` before in our docs, but I absolutely love it!

+
The {sandboxed-containers-operator} is tasked with managing the lifecycle of sandboxed containers on a cluster. It deals with operations, such as the installation and removal of sandboxed containers software and status monitoring.

> Kata Containers:: Kata Containers is a core upstream projects used to build {sandboxed-containers-first}. {sandboxed-containers-first} integrate Kata Containers with {product-title}.

Reviewer: ```suggestion
Kata Containers:: Kata Containers is a core upstream project that is used to build {sandboxed-containers-first}. {sandboxed-containers-first} integrate Kata Containers with {product-title}.
```

> KataConfig:: `KataConfig` objects represents configurations of sandboxed containers. They store information about the state of the cluster, such as the nodes on which the software is deployed.

Reviewer 4: s/represents/represent

> KataConfig:: `KataConfig` represents a configuration of sandboxed containers. Information about the state of the cluster, for example the nodes on which the software was deployed, is in the `KataConfig` objects.

Reviewer: ```suggestion
KataConfig:: `KataConfig` objects represents configurations of sandboxed containers. They store information about the state of the cluster, such as the nodes on which the software is deployed.
```

{op-system} extensions:: {op-system-first} extensions are a mechanism to install optional {product-title} software. The {sandboxed-containers-operator} uses this mechanism to deploy sandboxed containers on a cluster.

> Runtime class:: A `RuntimeClass` object describes which runtime can be used to run a given workload. A runtime class that is named `Kata` is installed and deployed by the {sandboxed-containers-operator}. The runtime class contains information about the runtime that describes resources that the runtime needs to operate, such as the pod overhead.

Reviewer 5: nit: The RuntimeClass is named `kata`

Reviewer 3: @bpradipt: Good catch ;-)

> Runtime class:: A `RuntimeClass` object describes which runtime can be used to run a given workload. A runtime class that is named `kata` is installed and deployed by the {sandboxed-containers-operator}. The runtime class contains information about the runtime that describes resources that the runtime needs to operate, such as the pod overhead.

Reviewer 4: Is "the pod overhead" clear enough? I've not run across this usage before and can't tell if it means "overhead" as in "above" or as in "debt".

Author: I believe it is in terms of debt. @jensfr could you clarify?

Reviewer 3: The term is a standard Kubernetes term with its own page. You could link to it: https://kubernetes.io/docs/concepts/scheduling-eviction/pod-overhead/

> Runtime class:: A RuntimeClass object describes which runtime can be used to run a given workload. A runtime class that is named `Kata` is installed and deployed by the {sandboxed-containers-operator}. The runtime class contains information about the runtime that describes resources that the runtime needs to operate, such as the pod overhead.

Reviewer: Guidelines tweak: 

```suggestion
Runtime class:: A `RuntimeClass` object describes which runtime can be used to run a given workload. A runtime class that is named `Kata` is installed and deployed by the {sandboxed-containers-operator}. The runtime class contains information about the runtime that describes resources that the runtime needs to operate, such as the pod overhead.
```

//Module included in the following assemblies:
//
// * sandboxed_containers/understanding_sandboxed_containers.adoc

[id="sandboxed-containers-common-terms_{context}"]

= Common terms

> .Sandbox

Reviewer: Thought: this would be a great place for [description lists](https://asciidoctor.org/docs/asciidoc-writers-guide/#description-lists).

Author: This is a great idea! I didn't think about description list. I think this is a good idea as well.

> A sandbox is a controlled environment where programs can run. Within the controlled environment, you can execute untested or untrusted programs without risking harm to the host machine or the operating system. A sandbox keeps your workloads isolated from the rest of the system.

Reviewer 6: >! execute _verb_
Use only if a simpler term, such as "run", "issue", "start", or "enter" is inappropriate.

> In the case of {sandboxed-containers-first}, sandboxing is achieved by running workloads in a different kernel using virtualization, providing enhanced control over the interactions between multiple workloads running on the same host.

Reviewer 6: >! using _noun_
To avoid ambiguity, replace this gerund with either "by using" or "that uses".

.Pod

> A pod is a kubernetes and OpenShift construct that represents resources where containers can be deployed. In other words, containers run inside pods. Pods are used for example to specify resources that can be shared between multiple containers.

Reviewer 6: We [almost never use ](https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#product-name-and-version)plain "OpenShift." This should probably be `{product-title}`.

> A pod is a kubernetes and OpenShift construct that represents resources where containers can be deployed. In other words, containers run inside pods. Pods are used for example to specify resources that can be shared between multiple containers.

Reviewer: +1. I zeroed in on "Kubernetes" too quickly.

In the case of {sandboxed-containers-first}, a pod is implemented as a virtual machine. Several containers can run in the same pod, and therefore in the same virtual machine.

> .OpenShift sandboxed containers Operator

Reviewer 6: +1 - you can (and should) use the attributes everywhere but in the `_topic_map.yml` file. I see a couple of OpenShift sandboxed containers and RHCOS references in headings that should be swapped to attributes.

> .OpenShift sandboxed containers Operator

Author: @maxwelldb + @kalexand-rh I was leaving it off from headings because of PV2 compliance. PV2 checklist: Do not use {variables} in title/chapter names or IDs

> .OpenShift sandboxed containers Operator

Reviewer 7: Yes, ignore the PV2 checklist for now. FWIW, that guidance is only temporary.

An operator is a software component that automates operations, automating actions that a human operator could do on the system.

The {sandboxed-containers} is tasked with managing the lifecycle of sandboxed containers on a cluster. It deals with operations such as sandboxed containers software installation and removal, and status monitoring.

.Kata Containers
Kata Containers is the name of one of the core upstream projects used in building {sandboxed-containers-first}. {sandboxed-containers-first} integrate Kata Containers with OpenShift.

.KataConfig

> `KataConfig` represents a configuration of sandboxed containers. Information about the state of the cluster, for example the nodes on which the software was deployed, can be found in the `KataConfig` objects.

Reviewer 6: You could tighten this a little.
```suggestion
`KataConfig` represents a configuration of sandboxed containers. Information about the state of the cluster, for example the nodes on which the software was deployed, is in the `KataConfig` objects.
```

.Red Hat CoreOS extensions

> Red Hat CoreOS (RHCOS) extensions are a mechanism to install optional OpenShift software. The {sandboxed-containers} uses this mechanism to deploy sandboxed containers on a cluster.

Reviewer 6: ```suggestion
{op-system-first} extensions are a mechanism to install optional {product-title} software. The {sandboxed-containers} uses this mechanism to deploy sandboxed containers on a cluster.
```

There are attributes for both first and subsequent use of RHCOS - {op-system-first} and {op-system}. Please use them throughout.

> Red Hat CoreOS (RHCOS) extensions are a mechanism to install optional OpenShift software. The {sandboxed-containers} uses this mechanism to deploy sandboxed containers on a cluster.

Reviewer: Yeah, I thought that these existed, but I couldn't find them in the common attributes file, so I assumed they'd been nixed. I blame my ophthalmologist appointment earlier in the day. :)
